sources/OpenMcdf.Extensions/OLEProperties/PropertyFactory.cs

```diff
@@ -412,26 +412,86 @@ public override string ReadScalarValue(System.IO.BinaryReader br)
                 uint size = br.ReadUInt32();
                 data = br.ReadBytes((int)size);
 
-                return Encoding.GetEncoding(codePage).GetString(data);
+                var result = Encoding.GetEncoding(codePage).GetString(data);
+                //result = result.Trim(new char[] { '\0' });
+
+                //if (this.codePage == CodePages.CP_WINUNICODE)
+                //{
+                //    result = result.Substring(0, result.Length - 2);
+                //}
+                //else
+                //{
+                //    result = result.Substring(0, result.Length - 1);
+                //}
+
+                return result;
             }
 
             public override void WriteScalarValue(BinaryWriter bw, string pValue)
             {
-                data = Encoding.GetEncoding(codePage).GetBytes(pValue);
-                uint dataLength = (uint)data.Length;
+                //bool addNullTerminator = true;
+
+                if (String.IsNullOrEmpty(pValue)) //|| String.IsNullOrEmpty(pValue.Trim(new char[] { '\0' })))
+                {
+                    bw.Write((uint)0);

```

Comment:
Does the zero length string case still need to be null terminated? (the docs at https://learn.microsoft.com/en-us/openspecs/office_file_formats/ms-oshared/fac324c9-ff39-442e-bd18-1a91a723a818 sound unclear to me when it says "SHOULD specify the number of characters in the value field including the terminating NULL character"

Reply:
No: if i'm reading the correctly, current ms-oleps states that if string is zero length, characters field shouldn't be present (2.5) so the zero length string IS a valid case imho

Reply:
Hmm, after reading the docs for CodePageString rather than Lpstr I was going to agree with you, but then I tried writing a .doc file with a zero length user defined property string using this branch, and when I tried looking at the properties with Windows Explorer, the whole of Explorer crashed :-( (Though the file displays correctly in Word)

Reply:
There's different verbiage here:  https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-oleps/a4c32611-5b79-4965-8f50-50639c138e16

That seems to say that if the length is zero, you write nothing.

> Characters (variable): If Size is zero, this field MUST be zero bytes in length.

Reply:
The behavior potentially depends on if you're treating the strings as ```CodePageString``` as per https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-oleps/a4c32611-5b79-4965-8f50-50639c138e16 or ```Lpstr``` as per https://learn.microsoft.com/en-us/openspecs/office_file_formats/ms-oshared/fac324c9-ff39-442e-bd18-1a91a723a818 - Microsoft don't make some of this stuff easy :-(

Reply:
No, some of these specs are really confusing to interpret.